application/src/main/java/org/togetherjava/tjbot/commands/BotCommandAdapter.java

```diff
@@ -54,17 +52,17 @@ protected BotCommandAdapter(CommandData data, CommandVisibility visibility) {
         this.data = Objects.requireNonNull(data, "The data shouldn't be null");
         this.visibility = Objects.requireNonNull(visibility, "The visibility shouldn't be null");
 
-        this.name = data.getName();
-        this.type = data.getType();
+        type = data.getType();
+        componentIdInteractor = new ComponentIdInteractor(data.getName());
     }
 
     @Override
     public final String getName() {
-        return name;
+        return componentIdInteractor.getName();
```

Comment:
same. i already redesigned. its not part of the component-id-interactors job to provide a name. it needs a name, but shouldnt provide it.

the componentIdInteractor does add prefixes, thats its job. therefore, the botcommandadapter doesnt have to think about this anymore since it just delegates.